tests/integration/components/amount-input/component-test.js

```diff
@@ -39,6 +39,43 @@ module('Integration | Component | amount-input', function(hooks) {
     assert.dom('input').hasAttribute('placeholder', '1.000.000')
   })
 
+  test('defaults are not used if named property declared even if undefined', async function(assert) {
+    await render(hbs`
+      <AmountInput
+        @currency={{this.isUndefined}}
+        @numberOfDecimal={{this.isUndefined}}
+        @placeholder={{this.isUndefined}}
+        @step={{this.isUndefined}}
+        @value={{this.value}}
+        @update={{fn (mut this.value)}} />
+    `)
+    assert.dom('.amount-input__currency').hasText('')
+    assert.dom('input').hasNoAttribute('placeholder')
+
+    // test numberOfDecimal
+    await fillIn('input', '10.726')
+    await blur('input')
+
+    assert.dom('input').hasValue('11')
```

Comment:
No test for `step` attribute

Triggering an arrow up doesn't work in the test, works when manually done.
```js
await focus('input');
await triggerKeyEvent('input', 'keydown', 'ArrowUp')
```